### DISCUSSION TITLE
LiveKit

### DISCUSSION THREAD

**Discussion by poberbeck** [coollabsio/coolify#4124] [Service Template Requests]
# Overview:

[LiveKit](https://livekit.io) is an open-source, scalable WebRTC server designed for building real-time audio and video applications. It offers client SDKs for multiple platforms, including JavaScript, Swift, Android, Flutter, React Native, Rust, Node.js, Python, and Unity, facilitating cross-platform development. ([GitHub](https://github.com/livekit/livekit))

# Key Features:

- Scalability: Supports large-scale deployments with thousands of concurrent participants.
- Low Latency: Utilizes WebRTC technology to ensure minimal latency for interactive communication.
- Flexibility: Provides client SDKs for multiple platforms, enabling diverse application development.
- Extensibility: Offers an architecture that allows easy integration with other services and customization to meet specific application requirements.

# Potential Use Cases:

- Video Conferencing: Develop your own platforms for virtual meetings, webinars, and online classes.
- Live Streaming: Create your own solutions for broadcasting live events, gaming streams, and interactive shows.
- Real-Time Collaboration: Build tools for collaborative work, including shared whiteboards and co-editing documents.

Docs for self-hosting can be found [here](https://docs.livekit.io/home/self-hosting/deployment/).

**Comment by reneil1337** [comment]
yes pleaaase

**Comment by ahmadaidil** [comment]
I need this!

**Comment by ayush-that** [comment]
Wish this existed :)

**Reply by mdcnick** [reply]
ill make it

**Reply by Joshokelola** [reply]
> ill make it

Hello, were you able to set it up?

**Reply by PSchartner123** [reply]
is there any update on this? would be crazy important for the voice ai bubble

**Comment by KazBrekker1** [comment]
Any efforts put into this?
I tired several times but I hit some obstacles when trying to include STUN/TURN (using coturn) and the ports exposed that are required for the server to run.

**Comment by Slawikus** [comment]
Would be great to have it!
